python/ray/data/grouped_dataset.py

```diff
@@ -144,6 +146,111 @@ def _aggregate_on(
         )
         return self.aggregate(*aggs)
 
+    def map_groups(
+        self,
+        fn: Union[CallableClass, Callable[[BatchType], BatchType]],
+        *,
+        compute: Optional[str] = None,
+        batch_format: str = "native",
+        **ray_remote_args
+    ) -> "Dataset[Any]":
+        """Apply the given function to each group of records of this dataset.
+
+        While the map_groups() is very flexibile, note that it comes with downsides:
+            - It may be slower than using more specific methods such as min(), max().
+            - It requires the size of each group fit on a single node.
+        So when specific methods are applicable, try to use them before reaching
+        map_groups().
+
+        This is a blocking operation.
+
+        Examples:
+            >>> # Compute an arbitrary function on each group.
+            >>> ray.data.range(100).groupby(lambda x: x % 3).map_groups(
+                    lambda x: [min(x) / max(x)])
+
+            >>> # Define a callable class that persists state across
+            >>> # function invocations for efficiency.
+            >>> class CachedModel:
+            ...    def __init__(self):
+            ...        self.model = init_model()
+            ...    def __call__(self, group):
+            ...        return self.model(group)
+
+            >>> # Apply the transform in parallel on GPUs. Since
+            >>> # compute="actors", the transform will be applied on an
+            >>> # autoscaling pool of Ray actors, each allocated 1 GPU by Ray.
+            >>> ds.map_groups(CachedModel, compute="actors", num_gpus=1)
+
+        Args:
+            fn: The function to apply to each group of records , or a class type
+                that can be instantiated to create such a callable.
+            compute: The compute strategy, either "tasks" (default) to use Ray
+                tasks, or "actors" to use an autoscaling Ray actor pool.
```

Comment:
The pydoc for the compute arg has been updated in master, make sure to copy from the latest version.

Reply:
Done.